app/models/entities/opportunity.rb

```diff
@@ -103,6 +103,14 @@ def weighted_amount
     ((amount || 0) - (discount || 0)) * (probability || 0) / 100.0
   end
 
+  def have_amount_to_display?
+    !!(
+      (amount && amount != 0) ||
```

Comment:
Its a bit annoying that all through the class we're doing ```amount || 0``` type logic to cater for this, rather than casting everything .to_i or .to_f as appropriate - here, the methods above, the views

Reply:
Raised #752 to clean that up outside of this PR